src/apps/answers/service.py

```diff
@@ -1212,12 +1212,28 @@ async def fill_last_activity(
         respondents: list[WorkspaceRespondent],
         applet_id: uuid.UUID | None = None,
     ) -> list[WorkspaceRespondent]:
-        respondent_ids = [respondent.id for respondent in respondents]
+        def get_subject_ids(r):
+            return list(map(lambda x: x.subject_id, respondent_item.details))
+
+        subjects_ids = []
+        for respondent_item in respondents:
+            subjects_ids = get_subject_ids(respondent_item)
+            subjects_ids += subjects_ids
         result = await AnswersCRUD(self.answer_session).get_last_activity(
-            respondent_ids, applet_id
+            subjects_ids, applet_id
         )
         for respondent in respondents:
-            respondent.last_seen = result.get(respondent.id)
+            curr_sub_ids = map(
```

Comment:
maybe, it is better to full words, so that everyone understands better. I assume here you mean `current_subject_ids` ?